### PR TITLE
test(text-area): Change tests to react-testing-library and improve coverage

### PR DESCRIPTION
**Pull request by stephanerangaya**
## Summary

Related to https://github.com/Workday/canvas-kit/issues/286

Converts test to RTL, improves coverage, adds a static states story for visual regression testing, and uses a more BDD (behavioral-driven) approach for unit tests.

## Checklist

- [x] branch has been rebased on the latest master commit
- [x] tests are changed or added
- [x] `yarn test` passes
- [x] all (dev)dependencies that the module needs is added to its `package.json`
- [x] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

### CODE REVIEW

File: modules/text-area/react/spec/TextArea.spec.tsx [hunk 1]
```diff
@@ -1,44 +1,81 @@
...
+    });
+
+    describe('wrapped in a FormField', () => {
+      test('should pass axe DOM accessibility guidelines', async () => {
+        const html = ReactDOMServer.renderToString(
+          <FormField label="My Field" inputId="my-text-area-field">
+            <TextArea id="my-text-area-field" onChange={cb} />
+          </FormField>
+        );
+        expect(await axe(html)).toHaveNoViolations();
```
Comment (NicholasBoll) [inline modules/text-area/react/spec/TextArea.spec.tsx]: I think this type of test should be moved to Cypress, right? Accessibility runs audits on our stories and not our unit tests.

This request can be part of our broader refactor of testing where component specifications are written and not part of this PR

Comment (anicholls) [reply modules/text-area/react/spec/TextArea.spec.tsx]: This may be good to include in the Cypress section of the testing docs

**Comment by stephanerangaya** [comment]
> I think for StaticStates to work, we need to import the styled function from labs core, i forgot about this!

They work I think, aren't they?

File: modules/text-area/react/spec/TextArea.spec.tsx [hunk 2]
```diff
@@ -1,44 +1,67 @@
...
+      expect(getByRole('textbox')).toHaveAttribute('data-propspread', attr);
+    });
+  });
+
+  describe('when provided an input ref', () => {
+    it('should set the ref to the input element', () => {
+      const ref: React.RefObject<HTMLInputElement> = React.createRef();
+      render(<TextArea inputRef={ref} id={id} />);
+      expect(ref.current).not.toBeNull();
+      expect(ref.current).toHaveAttribute('id', id);
```
Comment (NicholasBoll) [inline modules/text-area/react/spec/TextArea.spec.tsx]: I think the following should work:
```ts
expect(await getByRole('textbox')).toEqual(ref.current)
```

Also, as a side-note, I think it is safer for Testing Library APIs to be used with the promise API since rendering can be asynchronous. The synchronous API could break if we slightly change the implementation of components or React slightly changes timing (e.g. `useEffect` vs `componentDidMount` or concurrent mode)

**Comment by NicholasBoll** [comment]
@mannycarrera4 Do you want to approve the IconButton differences? I think they are from the IconButton changes, but I wanted to be sure.

The other change is new visual states for the text area. I'm looping in UX to make sure we got it right and that it makes sense.

File: modules/text-area/react/spec/TextArea.spec.tsx (hunk 2 above)
Comment (anicholls) [reply modules/text-area/react/spec/TextArea.spec.tsx]: @NicholasBoll Are you saying that we should aways be using `expect(await getBy*())` instead of `expect(getBy*())`?

Comment (NicholasBoll) [reply modules/text-area/react/spec/TextArea.spec.tsx]: I know a bit more about the React Testing Library APIs. There are `get*`, `query*`, and `find*` queries.

`find*` is most likely what we want for React because of timing. `find*` returns a promise that resolves when the element is found (retrying until it is found). If there is a delay in rendering something (hooks, concurrent mode, etc), `find*` will eventually find the element and return it for testing.

So I should update my previous code snippet:

```ts
expect(await findByRole('textbox')).toEqual(ref.current)
```

Comment (NicholasBoll) [reply modules/text-area/react/spec/TextArea.spec.tsx]: I'm not sure if `await find*` vs `get*` is strictly necessary. React Testing Library hooks into some testing internals to fix timing issues introduced with hooks. I just consider timing to be an implementation detail of the component that should not introduce failures or flakiness of tests.

Comment (anicholls) [reply modules/text-area/react/spec/TextArea.spec.tsx]: As discussed, we agreed to leave this change to a later PR because it will be an all-or-nothing change to our TL/R tests.
